src/librustdoc/html/static/js/search.js

```diff
@@ -684,39 +715,23 @@ window.initSearch = function(rawSearchIndex) {
                 fullId = ty.id;
 
                 returned = checkReturned(ty, output, true, NO_TYPE_FILTER);
-                if (output.name === "*" || returned) {
+                if (output.name === "*" || returned === 0) {
                     in_args = false;
                     var is_module = false;
 
                     if (input === "*") {
                         is_module = true;
                     } else {
-                        var allFound = true;
-                        for (it = 0, len = inputs.length; allFound && it < len; it++) {
+                        var allFound = 0;
+                        for (it = 0, len = inputs.length; allFound === 0 && it < len; it++) {
                             allFound = checkType(type, inputs[it], true);
                         }
                         in_args = allFound;
```

Comment:
This loop is pretty confusing to me. I tried to rewrite it with different variable names and a structure that makes its intended meaning clearer (at least I think so!).

```suggestion
                        var firstNonZeroDistance = 0;
                        for (var it = 0, len = inputs.length; it < len; it++) {
                            var distance = checkType(type, inputs[it], true);
                            if (distance > 0) {
                                firstNonZeroDistance = distance;
                                break;
                            }
                        }
                        in_args = firstNonZeroDistance;
```